src/sentry/static/sentry/app/views/settings/projectIncidentRules/ruleForm.tsx

```diff
@@ -77,6 +116,15 @@ class RuleForm extends React.Component<Props, State> {
   ): AlertRuleThresholdKey[AlertRuleThreshold] =>
     type === AlertRuleThreshold.RESOLUTION ? 'resolveThreshold' : 'alertThreshold';
 
+  /**
+   * Gets a reasonable period given a timewindow (in seconds)
+   *
+   * @param timeWindow The time window in seconds
+   * @return period The period string to use (e.g. 14d)
+   */
+  getPeriodForTimeWindow = (timeWindow: TimeWindow): string =>
```

Comment:
Does this type assertion make the second half of the `||` expression unreachable?

Reply:
I think it's because `TIME_WINDOW_TO_PERIOD` map is considered mutable so the compiler thinks it can return undefined -- will find a better way to handle this.